### PR TITLE
Mark fields that are not required as optional

### LINKED ISSUE #244: String values that are omittable should also be optional
Otherwise it's very hard to delete an existing value by doing an update.

### PR DESCRIPTION
**Pull request by jcoyne**
## Why was this change made?

this gives us a way to do an update where we clear a value without having to have all the values of a schema

Fixes #244

### CODE REVIEW

**Review by justinlittman — CHANGES_REQUESTED** [review]
Can you add a spec for generator?

**Comment by jcoyne** [comment]
@justinlittman you bet! I added one.

File: lib/cocina/generator/schema_base.rb [hunk 1]
```diff
@@ -21,12 +21,20 @@ def name
         key || schema_doc.name
       end
 
+      # Allows values to be not be provided. This allows smaller requests as not every field needs to be provided.
```
Comment (ndushay) [inline lib/cocina/generator/schema_base.rb:24]: "to be not be provided"?  change to "not to be provided"?

File: lib/cocina/generator/schema_base.rb [hunk 2]
```diff
@@ -21,12 +21,20 @@ def name
...
       def omittable
         return '' if required
 
         '.meta(omittable: true)'
       end
 
+      # Allows values to be set to nil. This is useful when doing an update and you want to clear out a value.
```
Comment (ndushay) [inline lib/cocina/generator/schema_base.rb:31]: "Allows non-required values to be set to nil" ???

**Review by ndushay — CHANGES_REQUESTED** [review]
Couple of comment quibbles;  o.w. looks okay to merge but I think 

Since this change impacts every model, I think it warrants running the integration tests before rolling it to prod.  (that is, after gem is released with this change, deploy to qa and run integration tests).

**Comment by jcoyne** [comment]
@ndushay thanks for the constructive suggestions. Changes made.

**Review by ndushay — APPROVED** [review]
approving as my comment quibbles were addressed;  I also don't want to block this while I'm out with the 2nd jab reaction.  I stand by my statement about testing this via integration tests before rolling it to prod.

**Pull request merged**